python-backend/app/api/mines/notification/resources/notification.py

```diff
@@ -0,0 +1,51 @@
+from flask_restplus import Resource
+from sqlalchemy_filters import apply_sort
+
+from app.extensions import api
+from app.api.utils.include.user_info import User
+from ....utils.access_decorators import requires_role_mine_view
+from ....utils.resources_mixins import UserMixin, ErrorMixin
+from ..models.notification import Notification
+from ...mine.models.mine import Mine
+
+
+class MineNotificationResource(Resource, UserMixin, ErrorMixin):
+    @api.doc(params={'mine_guid': 'Mine guid.'})
+    @requires_role_mine_view
+    def get(self):
+        user_name = User().get_user_username()
+        mine_query = Mine.query.filter_by( deleted_ind=False ).join(Notification).filter_by(idir=user_name)
+        sort_criteria = [{'model': 'Mine', 'field': 'mine_name', 'direction': 'asc'}]
+        mine_query = apply_sort(mine_query, sort_criteria)
+        mines = mine_query.all()
+        return {
+            'mines': list(map(lambda x: x.json_for_list(), mines)),
+        }
+
+    @api.doc(params={'mine_guid': 'Mine guid.'})
+    @requires_role_mine_view
+    def post(self, mine_guid=None):
+        if mine_guid:
+            try:
+                mine_subscription = Notification.create_subscription(mine_guid)
+                return {
+                    'mine_guid': str(mine_subscription.mine_guid),
+                }
+            except Exception:
```

Comment:
Catching all exceptions is considered a bad practice. We should be catching a specific issue here.

And, as a result of moving the errors up in the method as early exits, the `return`'d dict can be moved to the end

Reply:
If you have specific things to catch, please do, but as of some new PR's AssertionErrors that bubble up will be turned into 400's, we don't have to catch them individually. same with any other exception gets turned into a 500 unless you specify. 

This will be explain in developers RC soon, but we want to stop using return self.create_error_payload and just raise the appropriate werkzeug exception.